src/main/java/com/notably/commons/core/path/AbsolutePath.java

```diff
@@ -1,53 +1,113 @@
 package com.notably.commons.core.path;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
 
 import com.notably.commons.core.path.exceptions.InvalidPathException;
 
 /**
  * TODO: Add Javadoc
  */
 public class AbsolutePath implements Path {
+    public static final String INVALIDABSOLUTEPATH = "Invalid absolute path";
+    private final List<String> paths;
+
+
+    public AbsolutePath(String absolutePathString) {
+        this.paths = new ArrayList<>();
+        for (String obj : absolutePathString.split("/")) {
+            if (!obj.trim().equals("")) {
+                this.paths.add(obj.trim());
+            }
+        }
+    }
+
     /**
-     * TODO: Add Javadoc
+     * Check if string is a ValidAbsolutePath
+     * @param absolutePathString
+     * @return
      */
     public static boolean isValidAbsolutePath(String absolutePathString) {
-        // TODO: Add implementation
+        if (absolutePathString.charAt(0) == '/') {
+            if (absolutePathString.contains("../")) {
+                return false;
+            } else {
+                return true;
+            }
+        }
         return false;
     }
 
     /**
-     * TODO: Add Javadoc
+     * Creates an absolute path from a string.
+     * @param absolutePathString used to create an absolutePath.
+     * @return Absolute Path.
+     * @throws InvalidPathException if String provided is not a valid absolutePath.
      */
     public static AbsolutePath fromString(String absolutePathString) throws InvalidPathException {
-        // TODO: Add implementation
-        return new AbsolutePath();
+        if (isValidAbsolutePath(absolutePathString)) {
+            return new AbsolutePath(absolutePathString);
+        } else {
+            throw new InvalidPathException(INVALIDABSOLUTEPATH);
+        }
     }
 
     /**
-     * TODO: Add Javadoc
+     * Convert from relative path to absolute path.
+     * @param relativePath used to convert to absolute path.
+     * @param currentWorkingPath of the current working directory.
+     * @return the converted AbsolutePath.
+     * @throws InvalidPathException
      */
     public static AbsolutePath fromRelativePath(RelativePath relativePath,
             AbsolutePath currentWorkingPath) throws InvalidPathException {
-        // TODO: Add implementation
-        return new AbsolutePath();
+        List<String> temp = new ArrayList<>(relativePath.getComponents());
+        List<String> temp2 = new ArrayList<>(currentWorkingPath.getComponents());
+        temp2.addAll((temp));
+        return new AbsolutePath(String.join("/", temp2));
```

Comment:
Hmm, I don't think this will work. For example, if the `relativePath` contains `"./CS2103/../CS2103"` and the `currentWorkingPath` contains `"/root/school"`, you'll end up getting `"/root/school/./CS2103/../CS2103"`.

Reply:
Instead, I think you should do something similar to the following:

1. Initialize a `newComponents` variable of type `List<String>`.
2. Similar to what you have done, get the components of the `currentWorkingPath`. Append all the components to `newComponents`.
3. Loop through the components of the `relativePath` one by one.
   1. If you encounter a "." string, ignore it.
   2. If you encounter a ".." string, pop the last element of `newComponents`.
   3. Else, add the string to `newComponents`.
4. Then, lastly, create the new `AbsolutePath` using `newComponents`. Perhaps, instead of joining these `newComponents` into a string and call the default constructor (with the string argument), you'd want to create a new constructor that accepts a `List<String>` parameter.